.travis.yml

```diff
@@ -17,85 +12,76 @@ before_install:
 install:
   -
 
-env:
-  -JDK=Java8
-  -JDK=Java9
-  -JDK=Java10
-  -JDK=Java11-EA
-  -JDK=Java12-EA
-
 matrix:
   fast_finish: true
   include:
     - jdk: oraclejdk8
       env:
         - DESC="acceptance tests"
-        - JDK=Java8
         - CMD="mvn install --projects acceptance-tests --also-make --activate-profiles all"
 
     - jdk: oraclejdk8
       env:
         - DESC="findbugs"
-        - JDK=Java8
         - CMD="mvn install findbugs:check --projects '!scala-unit-tests,!jmh-scala-tests,!jmh-tests,!jcstress-tests,!p2-repository' --activate-profiles all -DskipTests=true"
+
     - jdk: oraclejdk8
       env:
         - DESC="checkstyle"
-        - JDK=Java8
         - CMD="mvn install checkstyle:check --activate-profiles all -DskipTests=true"
 
     - jdk: oraclejdk8
       env:
         - DESC="unit tests"
-        - JDK=Java8
         - CMD="mvn install"
 
     - jdk: oraclejdk8
       env:
         - DESC="compile performance-tests"
-        - JDK=Java8
         - CMD="mvn install --projects performance-tests --also-make --activate-profiles all -DskipTests=true"
 
     - jdk: oraclejdk9
       env:
-        - DESC="unit tests Java 9"
-        - JDK=Java9
+        - DESC="unit tests oraclejdk9"
         - CMD="mvn install"
 
-    - jdk: oraclejdk8
+    - jdk: openjdk10
+      env:
+      - DESC="unit tests openjdk10"
+      - CMD="mvn install"
+
+    - jdk: oraclejdk10
+      env:
+      - DESC="unit tests oraclejdk10"
+      - CMD="mvn install"
+
+    - jdk: openjdk11
       env:
-        - DESC="Java 10 regression"
-        - JDK=Java10
-        - CMD=""
+      - DESC="unit tests openjdk11"
+      - CMD="mvn install"
 
     - jdk: oraclejdk8
       env:
-        - DESC="Java 11 regression"
+        - DESC="unit tests Java11-EA"
         - JDK=Java11-EA
-        - CMD=""
+        - CMD="./.travis-command-ea-builds.sh"
 
     - jdk: oraclejdk8
       env:
-        - DESC="Java 12 regression"
+        - DESC="unit tests Java12-EA"
         - JDK=Java12-EA
-        - CMD=""
+        - CMD="./.travis-command-ea-builds.sh"
```

Comment:
This is the crucial change that I really wanted to do. Launching an external script interferes with logging. The entire command is evaluated, and then only when its finished will anything appear in the logs. I want to launch an external script only when necessary. And when we launch an external script, it can just be the regular old CMD.